salt/metalk8s/roles/bootstrap/local.sls

```diff
@@ -1,6 +1,8 @@
 include:
   - metalk8s.archives.mounted
+{%- if not pillar.skip_kubelet | default(False) %}
```

Comment:
Didn't we settle on having another "role" to only upgrade the Salt master? I don't know, maybe simply running `state.sls metalk8s.salt.master` from the upgrade script would be enough?

Reply:
(And repo, we want to upgrade repo as well at this point)

No strong opinion here for me either:
1- Do like this with pillar variable (I would just move it to `metalk8s.skip.kubelet`)
2- Create a new role state in bootstrap (e.g.: `metalk8s.roles.bootstrap.component`) that only do salt.master and repo and then include this state here
3- Call directly the right state in the upgrade script (but may be a bit harder to maintain in the futur)

NOTE: Also to me you also want to skip, at least, kubectl during upgrade of bootstrap (kubectl should be upgraded after apiserver same as kubelet)

Reply:
IMO, skipping (option 1) and listing which states to use (option 3) are equally hard to maintain, as they force some coupling between the script and the states.
I guess this leaves us with option 2 :slightly_smiling_face:

Reply:
Repositories are already upgraded as well as Salt master kubeconfig and certs because of the `metalk8s.orchestrate.bootstrap.pre-upgrade` right before, so we only need to run `metalk8s.salt.master.installed` to make it works.
Maybe it's better if the repositories are explicitly deployed in some new role as you suggested, instead of relying on some dependencies (certs > m2crypto > repo), I will change it.

Reply:
> Repositories are already upgraded as well as Salt master kubeconfig and certs because of the metalk8s.orchestrate.bootstrap.pre-upgrade right before

Are you sure ? We only update the repo config on the host but not the repo container itself

Reply:
My bad, you're right, it's even earlier in the process.
It's due to the iso-manager script.

Reply:
My bad, you're right, just the iso manager popuplate the registry and salt master with stuff from the new ISO, it does not upgrade yet the images, still the new content is accessible.